crates/handlers/src/graphql/model/upstream_oauth.rs

```diff
@@ -45,6 +49,41 @@ impl UpstreamOAuth2Provider {
     pub async fn client_id(&self) -> &str {
         &self.provider.client_id
     }
+
+    /// The human-readable name of the provider.
+    pub async fn human_name(&self) -> Option<&str> {
+        self.provider.human_name.as_deref()
+    }
+
+    /// UpstreamOAuth2Links associated with this provider for the current user.
+    pub async fn upstream_oauth2_links_for_user(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Result<Vec<UpstreamOAuth2Link>, async_graphql::Error> {
```

Comment:
Not convinced it is good to have it at this level. I would rather have a list on the `User` to fetch links, and have the frontend do the stitching together of both lists?